hydradx/other tests/test_exploits.py

```diff
@@ -0,0 +1,645 @@
+import copy
+import math
+
+from hypothesis import given, strategies as st, settings
+from mpmath import mp, mpf
+
+from hydradx.model.amm import omnipool_amm as oamm
+from hydradx.model.amm.agents import Agent
+from hydradx.model.amm.global_state import GlobalState
+from hydradx.model.amm.trade_strategies import constant_swaps, omnipool_arbitrage
+from hydradx.tests.strategies_omnipool import omnipool_reasonable_config, omnipool_config
+mp.dps = 50
+
+# @settings(max_examples=1)
+@given(
+    st.floats(min_value=0, max_value=0.10, exclude_min=True),
+    st.floats(min_value=0, max_value=0.01, exclude_min=True),
+    # st.floats(min_value=0.90, max_value=1.1)
+)
+def test_add_liquidity_exploit(lp_multiplier, trade_mult):
+    oracle_mult = 1.0
+    # lp_multiplier = 0.5
+    # trade_mult = 0.5
+
+    tokens = {
+        'HDX': {'liquidity': 44000000, 'LRNA': 275143},
+        'WETH': {'liquidity': 1400, 'LRNA': 2276599},
+        'DAI': {'liquidity': 2268262, 'LRNA': 2268262},
+        'DOT': {'liquidity': 88000, 'LRNA': 546461},
+        'WBTC': {'liquidity': 47, 'LRNA': 1145210},
+    }
+
+    prices = {tkn: tokens[tkn]['LRNA'] / tokens[tkn]['liquidity'] for tkn in tokens}
+    trade_size = tokens['DOT']['liquidity'] * trade_mult
+
+    init_oracle = {
+        'liquidity': {tkn: tokens[tkn]['liquidity'] for tkn in tokens},
+        'volume_in': {tkn: 0 for tkn in tokens},
+        'volume_out': {tkn: 0 for tkn in tokens},
+        'price': {tkn: oracle_mult * prices[tkn] for tkn in tokens},
+    }
+
+    omnipool: oamm.OmnipoolState = oamm.OmnipoolState(
+        tokens=tokens,
+        preferred_stablecoin='DAI',
+        oracles={'price': 19, 'volatility': 19},
+        last_oracle_values={
+            'price': copy.deepcopy(init_oracle),
+            'volatility': copy.deepcopy(init_oracle),
+        },
+        withdrawal_fee=True,
+        min_withdrawal_fee=0.0001,
+        lrna_fee_burn=1.0,
```

Comment:
We need to test with lrna_fee_burn = 0.5 since that's what will be happening in PROD

Reply:
done